core/vm/intpool.go

```diff
@@ -65,3 +71,54 @@ func (p *intPool) put(is ...*big.Int) {
 		p.pool.push(i)
 	}
 }
+
+const poolPoolLimit = 5
+const (
+	PoolPoolTypeMiner = iota
+	PoolPoolTypeImporter
+	PoolPoolTypeTracer
+	PoolPoolTypeRPC
+	PoolPoolTypes
+)
```

Comment:
I think you over complicated this. intPoolPool should only contain a `[]*intPool`, which you programatically cap to about 5. There's no need to have dedicated pools for different subsystems.

Reply:
The idea is to behave like `sync.Pool`, just to survive GC runs (which clear `sync.Pool`).